src/main/java/de/muenchen/dave/services/messstelle/BelastungsplanService.java

```diff
@@ -57,9 +60,22 @@ public BelastungsplanMessquerschnitteDTO ladeBelastungsplan(final List<TotalSumP
         belastungsplanMessquerschnitteDTO.setTotalPercentGv(calcPercentage(totalSumGv, totalSum));
         belastungsplanMessquerschnitteDTO.setTotalPercentSv(calcPercentage(totalSumSv, totalSum));
         belastungsplanMessquerschnitteDTO.setLadeBelastungsplanMessquerschnittDataDTOList(listBelastungsplanMessquerschnitteDTO);
+        if (options.getMessquerschnittIds().size() == 1) {
+            final var isKfzStelle = Objects.equals(options.getZeitauswahl(), "Spitzenstunde KFZ");
+            final var spitzenstunde = spitzenstundeService.calculateSpitzenstunde(options.getZeitblock(), intervals, isKfzStelle);
+            belastungsplanMessquerschnitteDTO.setStartUhrzeitSpitzenstunde(spitzenstunde.getStartUhrzeit());
+            belastungsplanMessquerschnitteDTO.setEndeUhrzeitSpitzenstunde(spitzenstunde.getEndeUhrzeit());
+        }
         return belastungsplanMessquerschnitteDTO;
     }
 
+    private static String getStrassennameFromMessquerschnitt(ReadMessstelleInfoDTO messstelle) {
+        if(messstelle.getMessquerschnitte().isEmpty()) {
```

Comment:
Bitte hier die CollectionUtils.isNotEmpty(messstelle.getMessquerschnitte()) verwenden. Da wird auch gleich noch auf null vorher geprüft.

```suggestion
        if (CollectionUtils.isNotEmpty(messstelle.getMessquerschnitte())) {
```